File: backend/phrase/models/phrase.py
from django.db import models
from user.models.user import User

# Create your models here.

class Phrase(models.Model):
    id = models.AutoField(primary_key = True)
    user = models.ForeignKey(User , on_delete = models.CASCADE)
    title = models.CharField(max_length = 30)
    description = models.TextField(max_length = 100)
    date = models.DateField()

    Romance = 'Romance'
    Fiction = 'Fiction'
    Politics = 'Politics'
    Fantasy = 'Fantasy'
    Social = 'Social'
    Other = 'Other'

    category_choices = [
        (Romance, Romance),
        (Fiction, Fiction),
        (Politics, Politics),
        (Fantasy, Fantasy),
        (Social, Social),
        (Other, Other),
    ]

    category = models.CharField(max_length = 30, choices = category_choices, default = Romance,)

    def __str__(self):
        return self.title

File: frontend/src/app/models/phrase.ts

export class Phrase{
    public id?: string;
    public user?: {};
    public title?: string;
    public description?: string;
    public date?: string;
    public category?: string;

    public constructor(id: string, user: string, title: string, description: string, date: string, category: string){
        this.id = id;
        this.user = user;
        this.title = title;
        this.description = description;
        this.date = date;
        this.category = category;    
    }
}